### PR TITLE
Fix some issues with golangci-lint, urllint, imagelint, and misspell

### PR DESCRIPTION
**Pull request by davidvonthenen**
## What this PR does / why we need it

This fixes a number of issues with `urllint`, `imagelint`, `golangci-lint` and `misspell`. They include:
- relocates `golangci-lint`, `urllint` and `imagelint` into subdirectories of `check` to help try to clean up and keep "like" things together (aka linting/checks)
- removes the need to build binaries for `urrlint` and `imagelint` which would leave things on these binaries on the filesystem to clean up
- removes the generic `bin` directory (technically any path with the substring `bin` in it) from the `.gitignore` which was probably used to prevent the `golangci-lint`, `misspell`, `urllint` and `imagelint` binaries from being checked in on accident
- fixed an issue with `urllint` which would error out on directories with an extension of `.yaml`:
```
[vonthd@rocky8 community-edition]$ make urllint
cd ./hack/check/urllinter && \
go run main.go -path=/home/vonthd/go/src/github.com/vmware-tanzu/community-edition -config=.urllintconfig.yaml -summary=true -details=Fail
2022/03/21 13:47:03 read /home/vonthd/go/src/github.com/vmware-tanzu/community-edition/addons/packages/local-path-storage/0.0.19/bundle/config/upstream/local-path-storage.yaml: is a directory
```
- added functionality needed for the relocation of the `imagelint` utility which now allows the utility to take in fully qualified paths to the `addons` folder
- fixed an issue in which the `misspell` and `golangci-lint` utilities would only ever be downloaded once because it was hidden behind the `bin` in `.gitignore`. Moved the download (and subsequent clean up) of the utility into individual respective `/tmp` folders. This would effectively mirror what a build system does anyways which is always download the latest utilities. The options are to: 1) download a static version of the utility through the ensure-deps mechanism, or 2) always download the utilities so we catch a build failure early since a build node always downloads them

## Which issue(s) this PR fixes

Verified:
- `make urllint` checks urls
- `make imagelint` checked images
- `make misspell` caught at least one spelling error
- `make check` which runs everything

## Special notes for your reviewer

NA

### CODE REVIEW

**Comment by dvonthenen** [comment]
Ok, I think this is ready now

**Comment by dvonthenen** [comment]
> helloworld-go.example.svc.cluster.local

I know what happens... the urllint also can't take in a fully qualified path which is why it isn't filtering based on the exclude list. Will fix that right now.

File: .gitignore [hunk 1]
```diff
@@ -25,7 +25,6 @@ hack/NEW_BUILD_VERSION
 cayman_trigger.txt
 artifacts
 artifacts-admin
-bin
```
Comment (stmcginnis) [inline .gitignore:28]: We may want to keep this, at least for now:

```
❯ git st
## fix/urllintbuild
?? hack/check/tools/bin/
?? hack/tools/
```

Comment (stmcginnis) [reply .gitignore:28]: Nevermind, old binaries I needed to clean up.

**Review by stmcginnis — COMMENTED** [review]
Looks like just a lint issue to address yet.

**Review by stmcginnis — APPROVED** [review]

**Pull request merged**